src/extensions/scratch3_wedo2/index.js

```diff
@@ -757,6 +779,20 @@ class WeDo2 {
         }
     }
 
+    /**
+     * Check the battery level on the WeDo 2.0. If the WeDo 2.0 has disconnected
+     * for some reason, the BLE socket will get an error back and automatically
+     * close the socket.
+     */
+    _checkBatteryLevel () {
+        this._ble.read(
+            BLEService.DEVICE_SERVICE,
+            BLECharacteristic.LOW_VOLTAGE_ALERT,
+            false,
+            this._onMessage // TODO: make optional
```

Comment:
I had to put this back in to not clear the `characteristicDidChangeCallback` in the BLE socket.

Reply:
Per discussion, lets instead leave this parameter null, and in ble.js (and bt.js?), check for null before trying to call the callback

Reply:
bt.js doesn't seem to need it because `_messageCallback` there seems to be required in the constructor.  Leaving bt.js as is for now with respect to this, and can handle if it comes up later.